lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts

```diff
@@ -619,6 +623,104 @@ class WebMapServiceCatalogItem
     }
   );
 
+  fetchPalettes(): Promise<string[]> {
+    return new Promise((resolve, reject) => {
+      if (!this.isThredds) {
+        resolve([]);
+      }
+
+      if (this.palettes.length > 0) {
+        resolve([...this.palettes]);
+      }
+
+      const extractedUrl = this.uri!.clone()
+        .setSearch({
+          service: "WMS",
+          version: this.useWmsVersion130 ? "1.3.0" : "1.1.1",
+          request: "GetMetadata",
+          item: "layerDetails",
+          layerName: this.layersArray[0]
+        })
+        .toString();
+
+      if (!extractedUrl) {
+        resolve([]);
+      }
+      const proxiedUrl = proxyCatalogItemUrl(this, extractedUrl!);
+
+      fetch(proxiedUrl, { method: "GET" })
+        .then((response) => response.json())
+        .then((data) => {
+          this.setTrait(CommonStrata.user, "palettes", data.palettes);
+          this.setTrait(
+            CommonStrata.user,
+            "colorScaleMinimum",
+            data.scaleRange[0]
+          );
+          this.setTrait(
+            CommonStrata.user,
+            "colorScaleMaximum",
+            data.scaleRange[1]
+          );
+          this.setTrait(
+            CommonStrata.user,
+            "defaultPalette",
+            data.defaultPalette
+          );
+          this.setTrait(
+            CommonStrata.user,
+            "noPaletteStyles",
+            data.noPaletteStyles
+          );
+          this.setTrait(CommonStrata.user, "showPalettes", true);
+          resolve(data.palettes);
```

Comment:
The `user` stratum should only be used for user-driven changes to traits. The current `fetchPalettes` implementation makes changes to the `user` stratum - this means that when a share-link is generated, the palettes will be included. 

Can you please move the `fetchPalettes` logic into a new `LoadableStratum` (you can call it `NcMWSGetMetadataStratum`) ?